src/planning/strategies/convert-constraints-to-connections.ts

```diff
@@ -194,9 +194,9 @@ export class ConvertConstraintsToConnections extends Strategy {
                 let recipeHandleConnection = recipeParticle.connections[connection];
                 if (recipeHandleConnection == undefined) {
                   recipeHandleConnection =
-                      recipeParticle.addConnectionName(connection);
+                      recipeParticle.addConnectionName(connection) as HandleConnection;
```

Comment:
nit - why is this needed, `addConnectionName` already returns a `HandleConnection`?

Reply:
I think I added that first (when the type of recipeParticle wasn't known) but you're right, it isn't necessary now. Will remove in a follow-up PR

Reply:
https://github.com/PolymerLabs/arcs/pull/6135